src/Models/Template.php

```diff
@@ -19,37 +19,52 @@ protected static function booted()
 
     public static function content(Model $model, string $field)
     {
-        if ($model instanceof Product) {
-            $type = 1;
-            $replaces = [
-                '[product_name]'  => $model->name,
-                '[category_name]' => '',
-            ];
-        }
-
-        if ($model instanceof Category) {
-            $type = 2;
-            $replaces = [
-                '[category_name]' => $model->name,
-            ];
-        }
+        $type = self::getModelType($model);
 
-        if (!isset($type)) {
+        if (!$type) {
             return;
         }
 
         $value = optional(self::firstWhere('rule_type', $type))->{$field};
+
+        $replaces = [];
+        foreach (self::getReplaceKeys($value) as $key) {
+            $replaces[$key] = self::getReplaceValue($model, $key);
+        }
+
         $value = strtr($value, array_merge($replaces ?? [], [
             '[store_name]'          => Rapidez::config('general/store_information/name'),
             '[store_phone]'         => Rapidez::config('general/store_information/phone'),
             '[store_email]'         => Rapidez::config('trans_email/ident_general/email'),
             '[store_street_line_1]' => Rapidez::config('general/store_information/street_line1'),
             '[store_street_line_2]' => Rapidez::config('general/store_information/street_line2'),
             '[store_city]'          => Rapidez::config('general/store_information/city'),
-            '[store_postcode]'      => Rapidez::config('general/store_information/postcode'),
+            '[store_postcode]'      => Rapidez::config('general/store_information/postcode')
         ]));
 
         // Remove double spaces
         return preg_replace('/\s+/', ' ', $value);
     }
+
+    protected static function getModelType(Model $model) {
+        return match (true) {
+            $model instanceof Product => 1,
+            $model instanceof Category => 2
+        };
+    }
+
+    protected static function getReplaceKeys(string $value) {
+        preg_match_all('/\[(.*?)\]/', $value, $matches);
+
+        return $matches[0];
+    }
+
+    protected static function getReplaceValue(Model $model, string $replaceKey) {
+        $replace = match (true) {
+            $model instanceof Product => str_replace('product_', '', $replaceKey),
+            $model instanceof Category => str_replace('category_', '', $replaceKey)
+        };
+
+        return $model->{str_replace(['[', ']'], '', $replace)} ?? null;
```

Comment:
This may have some security implications if some fields should and have been hidden using the "hidden" array.
Accessing this as an array, or calling `$model->getAttributeValue(str_replace(['[', ']'], '', $replace))` should help lessen this

Reply:
Thanks @indykoning! I've added this in https://github.com/rapidez/mirasvit-advanced-seo-suite/pull/7.